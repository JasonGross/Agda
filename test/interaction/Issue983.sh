
top=Issue983
bad=Issue983-Bad
lib=Issue983-Lib

pipe=$top.pipe

rm -f $pipe
mkfifo $pipe

# Forces the pipe to stay open. Otherwise it's closed by echo the first time we
# write to it.
sleep 2 > $pipe &

agda --interaction < $pipe 2>&1 &

function cmd {
  echo "IOTCM \"$2.agda\" None Indirect (Cmd_$1 \"$2.agda\" $3)" > $pipe
}

# The library is empty
printf "\n" > $lib.agda

# So this won't work
printf "open import $lib\n"        > $bad.agda
printf "Bad = $lib.NonExisting\n" >> $bad.agda

# Load the library.
cmd load $lib []

# Now load $top. This jumps to the error in $bad.
cmd load $top []

# Loading the highlighting info for $bad. This looks in the moduleToSource map
# for $lib which should not cause an internal error!
cmd load_highlighting_info $bad

sleep 0.1
rm -f $pipe $lib.agda $lib.agdai $bad.agda

