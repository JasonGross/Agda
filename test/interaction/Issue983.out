Agda2> (agda2-status-action "")
(agda2-info-action "*Type-checking*" "" nil)
(agda2-highlight-clear)
(agda2-info-action "*Type-checking*" "Checking Issue983-Lib (Issue983-Lib.agda).\n" t)
(agda2-info-action "*Type-checking*" "Finished Issue983-Lib.\n" t)
(agda2-status-action "Checked")
(agda2-info-action "*All Goals*" "" nil)
((last . 1) . (agda2-goals-action '()))
Agda2> (agda2-status-action "")
(agda2-info-action "*Type-checking*" "" nil)
(agda2-highlight-clear)
(agda2-info-action "*Type-checking*" "Checking Issue983 (Issue983.agda).\n" t)
(agda2-info-action "*Type-checking*" " Checking Issue983-Bad (Issue983-Bad.agda).\n" t)
(agda2-info-action "*Error*" "Issue983-Bad.agda:2,7-31\nNot in scope:\n  Issue983-Lib.NonExisting\n  at Issue983-Bad.agda:2,7-31\nwhen scope checking Issue983-Lib.NonExisting" nil)
((last . 3) . (agda2-goto '("Issue983-Bad.agda" . 32)))
(agda2-highlight-load-and-delete-action)
(agda2-status-action "")
Agda2> (agda2-info-action "*Error*" "\nAn internal error has occurred. Please report this as a bug.\nLocation of the error:\nsrc/full/Agda/Interaction/InteractionTop.hs:852" nil)
(agda2-highlight-load-and-delete-action)
(agda2-status-action "")
Agda2> 